pkg/apis/serving/v1/configuration_types.go

```diff
@@ -105,3 +108,13 @@ type ConfigurationList struct {
 
 	Items []Configuration `json:"items"`
 }
+
+// GetTypeMeta retrieves the ObjectMeta of the Configuration. Implements the KRShaped interface.
+func (t *Configuration) GetTypeMeta() *metav1.TypeMeta {
```

Comment:
Any reason why `KRShaped` doesn't use https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1?tab=doc#Type

Then you get drop the `GetTypeMeta` methods

Reply:
Likewise instead of the `ObjectMetaAccessor` here: https://github.com/knative/pkg/blob/master/apis/duck/v1/kresource_type.go#L31

Could use https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#Object

Reply:
We can drop the accessor down to Object, but I'm having trouble with Type. Weirdly TypeMeta does not appear to implement Type as expected.